diffcp/cone_program.py

```diff
@@ -309,6 +291,23 @@ def derivative(dA, db, dc, **kwargs):
            NumPy arrays dx, dy, ds, the result of applying the derivative
            to the perturbations.
         """
+        Q = sparse.bmat([
+            [None, A.T, np.expand_dims(c, - 1)],
+            [-A, None, np.expand_dims(b, -1)],
+            [-np.expand_dims(c, -1).T, -np.expand_dims(b, -1).T, None]
+        ])
+
+        D_proj_dual_cone = _diffcp.dprojection(v, cones_parsed, True)
+        if mode == "dense":
+            Q_dense = Q.todense()
+            M = _diffcp.M_dense(Q_dense, cones_parsed, u, v, w)
+            MT = M.T
+
+        pi_z = pi(z, cones)
+
+        M = _diffcp.M_operator(Q, cones_parsed, u, v, w)
```

Comment:
Are these lines supposed to be in an `else` clause (they overwrite `M` and `MT` from lines 303-34)?